tests/insert_coins.js

```diff
@@ -0,0 +1,130 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+
+const keyring = new Keyring({ type: 'sr25519' });
+
+const fs = require('fs');
+
+// Create a promise API instance of the passed in node address.
+async function createPromiseApi(nodeAddress) {
+    const wsProvider = new WsProvider(nodeAddress);
+    const api = await new ApiPromise({ 
+        provider: wsProvider
+    });
+    await api.isReady;
+    return api;
+}
+const mints = [];
+
+async function main() {
+    let nodeAddress = "ws://127.0.0.1:9801";
+    
+    const api = await createPromiseApi(nodeAddress);
+    let sender = keyring.addFromMnemonic("bottom drive obey lake curtain smoke basket hold race lonely fit walk//Alice");
+    let receiver = keyring.addFromMnemonic("bottom drive obey lake curtain smoke basket hold race lonely fit walk//Bob");
+
+    const mints_file = '/home/georgi/Desktop/workspace/Manta/precomputed-15k-iterations/precomputed_mints_v3-5';
+    const transfers_file = '/home/georgi/Desktop/workspace/Manta/precomputed-15k-iterations/precomputed_transfers_v3-5';
+    const reclaims_file = '/home/georgi/Desktop/workspace/Manta/precomputed-15k-iterations/precomputed_reclaims_v3-5';
+
+    let mints_offset = 2;
+    let transfers_offset = 4;
+    let reclaims_offset = 4;
+    let total_iterations = 15000;
+    let collected_for_batch = 0;
+
+    let mint_size = 552;
+    var mints_buffer = Buffer.alloc(mint_size * total_iterations);
+    fs.open(mints_file, 'r',  async  function(status, fd1) {
```

Comment:
This looks like http://callbackhell.com/

Why not use promise API for fs access, ur in a `async fn` anyway

`var fd1 = await fs.open;`
havent tested the exact syntax, but there has to be a promise API for the fs module

Reply:
https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api

Reply:
Should be done now.